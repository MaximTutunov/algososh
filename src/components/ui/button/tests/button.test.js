import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "../button";

const buttonText = "кнопка с текстом";

describe("тестирование кнопки Button", () => {
  it("рендер кнопки Button  с текстом", () => {
    const tree = renderer.create(<Button text={buttonText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер кнопки Button без текста", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер кнопки Button заблокированной disabled", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("рендер кнопки Button с индикацией загрузки isLoader", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("вызов коллбека по клику на кнопку Button", () => {
    const fn = jest.fn();
    render(<Button text={buttonText} onClick={fn} />);
    const button = screen.getByText(buttonText);
    fireEvent.click(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
