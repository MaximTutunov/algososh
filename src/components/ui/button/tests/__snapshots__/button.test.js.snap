// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестирование кнопки Button рендер кнопки Button  без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`тестирование кнопки Button рендер кнопки Button  заблокированной disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`тестирование кнопки Button рендер кнопки Button  с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    кнопка с текстом
  </p>
</button>
`;

exports[`тестирование кнопки Button рендер кнопки Button без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`тестирование кнопки Button рендер кнопки Button заблокированной disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`тестирование кнопки Button рендер кнопки Button с индикацией загрузки isLoader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
